src/test/librados/c_write_operations.cc

```diff
@@ -259,15 +259,18 @@ TEST(LibRadosCWriteOps, CmpExt) {
   ASSERT_EQ(0, memcmp("five", hi, sizeof(hi)));
 
   // compare and bail before write due to mismatch
-  val = 0;
-  op = rados_create_write_op();
-  ASSERT_TRUE(op);
-  rados_write_op_cmpext(op, "four", 4, 0, &val);
-  rados_write_op_write(op, "six ", 4, 0);
-  ASSERT_EQ(-MAX_ERRNO - 1, rados_write_op_operate(op, ioctx, "test", NULL, 0));
-
-  ASSERT_EQ(-MAX_ERRNO - 1, val);
-
+  // do it 1000 times to make sure we are hitting
+  // some socket injection
+  for (auto i = 0; i < 1000; ++i) {
+    val = 0;
+    op = rados_create_write_op();
+    ASSERT_TRUE(op);
+    rados_write_op_cmpext(op, "four", 4, 0, &val);
+    rados_write_op_write(op, "six ", 4, 0);
+    std::string const s = "test_" + std::to_string(i);
+    ASSERT_EQ(-MAX_ERRNO - 1, rados_write_op_operate(op, ioctx, s.c_str(), NULL, LIBRADOS_OPERATION_RETURNVEC));
```

Comment:
nit: too long line.

nit: `NULL` instead of `nullptr` in changed code?

ACK for `LIBRADOS_OPERATION_RETURNVEC`.

Reply:
all other calls to rados_write_op_operate using NULL and not nullptr, do you think we should consider changing it?

Reply:
I won't run atilt for that :-).